db/log_writer.cc

```diff
@@ -85,10 +100,31 @@ IOStatus Writer::AddRecord(const Slice& slice) {
     assert(static_cast<int64_t>(kBlockSize - block_offset_) >= header_size);
 
     const size_t avail = kBlockSize - block_offset_ - header_size;
+
+    // Compress the record
+    if (compress_ && (compress_start || left == 0)) {
```

Comment:
This is kinda confusing. IIUC, we try to compress as much as will fit in a block. However, the first physical record for the logical record, and the start of the block may not be aligned. So the compressed buffer may be split into multiple physical records. Is that correct? Would the records have a dependency?

It might be simpler and easier to reason about if we have nested while loops.

Reply:
Yes, that is correct. There is no dependency other than all the compressed records need to be uncompressed in order to recover the original record. I tried changing the compress API to pass the available physical block size and it doesn't make a difference.

I can add some comments to make it more readable. The advantage of having a single loop is that there's less code required for compress vs uncompressed. Once a chunk is available, compressed or not, the rest of the code to generate the physical record is the same.

Reply:
Added comments.